feat(gh-pages) add a customizable index page + cleanup leftovers

dduportal opened the pull request:
This PR introduces the following changes (that I just applied on my own "slides" repository):

- Introduces an index page listing all the presentation. This page can always be changed afterwards with no impact, by editing it in your `gh-pages` branch directly.
  - Expected to be at https://jmmeessen.github.io/slides (instead of the current "tests slides" page)
  - If you want to check the look and feel on my own page: https://dduportal.github.io/slides/
  - Includes a custom CSS for bootstrap basic rendering (full W3C compliant)
  - Includes a JSON file to generate the list of links (incoming feature to automatically add a presentation when generated from GHA, but not finished yet)
  - Custom favico.ico :)
- Cleanup of leftovers from the old Travis CI system:
  - Fixed the qrcode on the JW EU 2019 presentation
  - custom favicon ;)
  - remove unused images and links

jmMeessen commented:
Thanks a lot for the help 👍

jmMeessen commented:
@dduportal There seems to be a glitch with this PR.

The "build" Github action fails with a not very explicit error message.

---
./custom.css
./favicon.ico
Error: Process completed with exit code 1.
---

At first sight, it is happening with this command:
`Run touch artifact.tar && tar --dereference --hard-dereference -cvf artifact.tar --exclude=.git --exclude=artifact.tar .`

dduportal commented:
@jmMeessen async. information:

- Problem fixed by the commit https://github.com/jmMeessen/slides/commit/1d0dfa3399091ef3608997269abd91c049f0d4d8 that I directly pushed (as a "hotfix") to your gh-pages , as you can see the result if now correct: https://github.com/jmMeessen/slides/actions/runs/2147174936

- Explanation: I'm the culprit, because I deleted the directory `./fonts` without fixing the broken symlinks in `jw-eu-2019`. That is the goal of the fix commit

- How to analyse such GitHub action behavior?
  - Step1: check the error (like you did): 
    - https://github.com/jmMeessen/slides/runs/5963636973?check_suite_focus=true#step:3:687 is the log that you pointed on your previous message.
    - Your analyse of the culprit command is correct: you provided the correct one.
  - Step 2: let's reproduce locally. The previous steps in the GitHub actions workflow are "Set Up" and "Checkout": it is quite easy.
    - "Set Up": The log of the "Set Up" step indicates that the environment is "Ubuntu 20.04 LTS": https://github.com/jmMeessen/slides/runs/5963636973?check_suite_focus=true#step:1:4. As the failing command uses only "usual" command lines (`touch`, `&&` which require a shell, wether `sh` or `bash` and `tar`), a docker image `ubuntu:20.04` should suffice
    - "Check out". I checked out the branch `gh-pages` of your repository. If you want to practise, you'll have to use the commit https://github.com/jmMeessen/slides/commit/30627020384b61c17d4fa9bb3a66a17c6a1189b3 that failed (as the next commit that I pushed, new HEAD of the branch,  is fixing the error)

```shell
# Checkout Step
$ git fetch --all --prune && git checkout 30627020384b61c17d4fa9bb3a66a17c6a1189b3

# Set Up step: share the current directory into the container
$ docker run --rm -ti -v $(pwd):$($pwd) -w $(pwd) --entrypoint=bash ubuntu:20.04
Unable to find image 'ubuntu:20.04' locally
20.04: Pulling from library/ubuntu
e0b25ef51634: Pull complete 
Digest: sha256:9101220a875cee98b016668342c489ff0674f247f6ca20dfc91b91c0f28581ae
Status: Downloaded newer image for ubuntu:20.04
# Reproduction
root@021e610ff21f:/Users/dadou/workspace/slides# touch artifact.tar && tar --dereference --hard-dereference -cvf artifact.tar --exclude=.git --exclude=artifact.tar .
# .... A bunch of logs ending with the error
```

Now that I could reproduce, I checked the (loooong) logs and some lines caught my eye:

```text
# ...
./devoxx-fr-2022/media/captain-iglo-movie.png
tar: ./jw-eu-2019/fonts/fontawesome-webfont.eot: File removed before we read it
tar: ./jw-eu-2019/fonts/FontAwesome.otf: File removed before we read it
tar: ./jw-eu-2019/fonts/fontawesome-webfont.svg: File removed before we read it
tar: ./jw-eu-2019/fonts/fontawesome-webfont.woff2: File removed before we read it
tar: ./jw-eu-2019/fonts/fontawesome-webfont.ttf: File removed before we read it
tar: ./jw-eu-2019/fonts/fontawesome-webfont.woff: File removed before we read it
./devoxx-fr-2022/media/cloudbees-logo.svg
# ...
```

If you check these files, you see there are broken symlinks:

```shell
$ ls -l ./jw-eu-2019/fonts/fontawesome-webfont.svg
lrwxr-xr-x  1 dadou  staff  35 Apr 11 10:04 ./jw-eu-2019/fonts/fontawesome-webfont.svg -> ../../fonts/fontawesome-webfont.svg
$  ls -l fonts/fontawesome-webfont.svg
ls: fonts/fontawesome-webfont.svg: No such file or directory
```

I fixed these links with the following onle-liner (but could be done manually given there were only a few):

```bash
for link in $(find . -type l);do echo "==$link"; f="$(basename "$link")"; test -f "master/fonts/$f" && unlink "$link" && cp "master/fonts/$f" "$link";done
```

Et voilà :)
